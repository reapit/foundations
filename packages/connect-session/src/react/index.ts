import React from 'react'
import { ReapitConnectHook, ReapitConnectSession } from '../types'
import { ReapitConnectBrowserSession } from '../browser'

export const useReapitConnect = (reapitConnectBrowserSession: ReapitConnectBrowserSession): ReapitConnectHook => {
  const [connectSession, setConnectSession] = React.useState<ReapitConnectSession | null>(null)

  React.useEffect(() => {
    const connectGetSession = async () => {
      const session = await reapitConnectBrowserSession.connectSession()
      if (session) {
        setConnectSession(session)
      }
    }

    connectGetSession()
  }, [])

  const connectAuthorizeRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectAuthorizeRedirect()
  }, [])

  const connectLoginRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLoginRedirect()
  }, [])

  const connectLogoutRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLogoutRedirect()
  }, [])

  const connectIsDesktop = reapitConnectBrowserSession.connectIsDesktop

  const connectHasSession = reapitConnectBrowserSession.connectHasSession

  return {
    connectSession,
    connectAuthorizeRedirect,
    connectLoginRedirect,
    connectLogoutRedirect,
    connectIsDesktop,
    connectHasSession,
  }
}
