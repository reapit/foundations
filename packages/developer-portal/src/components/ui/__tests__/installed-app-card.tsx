import * as React from 'react'
import { shallow } from 'enzyme'

import InstalledAppCard, { InstalledAppCardProps } from '../installed-app-card'

const props: InstalledAppCardProps = {
  app: {
    id: '09043eb8-9e5e-4650-b7f1-f0cb62699027',
    developerId: '28c9ea52-7f73-4814-9e00-4e3714b8adeb',
    name: 'test',
    summary:
      'nXXT2zaK807ysWgy8F0WEhIYRP3TgosAtfuiLtQCImoSx0kynxbIF0nkGHU36Oz13kM3DG0Bcsi' +
      'cr8L6eWFKLBg4axlmiOEWcvwHAbBP9LRvoFkCl58k1wjhOExnpaZItEyOT1AXVKv8PE44aMGtVz',
    developer: "Pete's Proptech World Ltd",
    homePage: 'http://google.com/abc',
    iconUri: 'https://reapit-app-store-app-media.s3.eu-west-2.amazonaws.com/d10e790c-2bf2-40ae-9c43-82c1534bde31.png',
    links: [
      {
        rel: 'self',
        href: 'http://platformdemo.reapit.net/marketplace/apps/09043eb8-9e5e-4650-b7f1-f0cb62699027',
        action: 'GET',
      },
      {
        rel: 'developer',
        href: 'http://platformdemo.reapit.net/marketplace/developers/28c9ea52-7f73-4814-9e00-4e3714b8adeb',
        action: 'GET',
      },
    ],
  },
  onClick: jest.fn(),
}

describe('InstalledAppCard', () => {
  const wrapper = shallow(<InstalledAppCard {...props} />)
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
