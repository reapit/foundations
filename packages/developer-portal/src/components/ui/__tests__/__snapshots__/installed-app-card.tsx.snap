// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstalledAppCard should match a snapshot 1`] = `
<div
  data-test-app-id="09043eb8-9e5e-4650-b7f1-f0cb62699027"
  data-test-app-name="test"
>
  <Component
    heading="test"
    image={
      <img
        alt="test"
        className="image"
        onError={[Function]}
        src="https://reapit-app-store-app-media.s3.eu-west-2.amazonaws.com/d10e790c-2bf2-40ae-9c43-82c1534bde31.png"
      />
    }
    onClick={[MockFunction]}
    subHeading="Pete's Proptech World Ltd"
  />
</div>
`;
